crates/dids/src/method/dht/mod.rs

```diff
@@ -0,0 +1,126 @@
+mod bep44;
+mod convert;
+pub mod registered_types;
+
+use std::sync::Arc;
+
+use crypto::Curve;
+use keys::key_manager::KeyManager;
+use reqwest::blocking::Client;
+
+use crate::{
+    bearer::BearerDid,
+    document::{Document, Service, VerificationMethod},
+    identifier::Identifier,
+};
+
+use self::{
+    bep44::encode_bep44_message, convert::document_packet::document_to_packet,
+    registered_types::RegisteredDidType,
+};
+
+use super::{Create, MethodError};
+
+const DEFAULT_GATEWAY_URL: &str = "https://diddht.tbddev.org";
+
+/// Concrete implementation for a did:dht DID
+pub struct DidDht;
+
+/// Options that can be used to create a did:dht DID
+pub struct DidDhtCreateOptions {
+    pub publish: bool,
+    // TODO: gatewayUri - default 'https://diddht.tbddev.org'
+    pub also_known_as: Option<Vec<String>>,
+    pub controller: Option<Vec<String>>,
+    pub service: Option<Vec<Service>>,
+    pub registered_type: Option<Vec<RegisteredDidType>>,
+    pub verification_methods: Option<Vec<VerificationMethod>>,
+}
+
+impl Create<DidDhtCreateOptions> for DidDht {
+    fn create(
+        key_manager: Arc<dyn KeyManager>,
+        options: DidDhtCreateOptions,
+    ) -> Result<BearerDid, MethodError> {
+        // Generate private and public keypair
+        let key_alias = key_manager.generate_private_key(
+            Curve::Ed25519,
+            Some("0".to_string()),
+        )?;
+        let public_key = key_manager.get_public_key(&key_alias)?;
+
+        // Create did uri
+        let public_jwk = public_key.jwk()?;
+        let jwk_string = serde_json::to_string(public_jwk.as_ref()).map_err(|_| {
+            MethodError::DidCreationFailure("failed to serialize public jwk".to_string())
+        })?;
+        let suffix = zbase32::encode_full_bytes(jwk_string.as_bytes());
+        let uri = format!("did:dht:{}", suffix);
+        let identifier = Identifier::parse(&uri).map_err(|e| {
```

Comment:
you may want to inline this, like so 

```suggestion
        let identifier = Identifier::parse(&format!("did:dht:{}", suffix)).map_err(|e| {
```

...and subsequently use the `identifier.uri` as the source of truth, that way we're fixing the `Identifier` if we discover issues, which will propagate to all uses of it